src/benchmark.cc

```diff
@@ -120,148 +109,118 @@ static const int kRangeMultiplier = 8;
 static const size_t kMaxFamilySize = 100;
 static const size_t kMaxIterations = 1000000000;
 
-bool running_benchmark = false;
+} // end namespace
 
-// Global variable so that a benchmark can cause a little extra printing
-std::string* GetReportLabel() {
-    static std::string label GUARDED_BY(GetBenchmarkLock());
-    return &label;
-}
+namespace internal {
 
-// Global variable so that a benchmark can report an error as a human readable
-// string. If error_message is null no error occurred.
-#if defined(_MSC_VER) && _MSC_VER <= 1800
-typedef char* error_message_type;
-#else
-typedef const char* error_message_type;
-#endif
+// NOTE: This is a dummy "mutex" type used to denote the actual mutex
+// returned by getBenchmarkMutex(). This is only used to placate the thread
+// safety warnings by giving the return of GetBenchmarkLock() a name.
+struct CAPABILITY("mutex") BenchmarkLockType {};
+BenchmarkLockType BenchmarkLockVar;
 
-static std::atomic<error_message_type> error_message = ATOMIC_VAR_INIT(nullptr);
+class ThreadManager {
+public:
+    ThreadManager(int num_threads)
+        : alive_threads_(num_threads),
+          start_stop_barrier_(num_threads)
+    {
+    }
 
-// TODO(ericwf): support MallocCounter.
-//static benchmark::MallocCounter *benchmark_mc;
+    Mutex& getBenchmarkMutex() const
```

Comment:
I'll do it right now if you clarify what the style should be (Other than FunctionNameAllUpper).